x-pack/plugins/apm/public/components/shared/TransactionDurationAnomalyAlertTrigger/index.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { EuiText, EuiSelect, EuiExpression } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { ALERT_TYPES_CONFIG } from '../../../../common/alert_types';
+import { ALL_OPTION, useEnvironments } from '../../../hooks/useEnvironments';
+import { useServiceTransactionTypes } from '../../../hooks/useServiceTransactionTypes';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { ServiceAlertTrigger } from '../ServiceAlertTrigger';
+import { PopoverExpression } from '../ServiceAlertTrigger/PopoverExpression';
+import { SelectAnomalySeverity } from './SelectAnomalySeverity';
+
+interface Params {
+  windowSize: number;
+  windowUnit: string;
+  serviceName: string;
+  transactionType: string;
+  environment: string;
+  anomalyScore: 0 | 25 | 50 | 75;
+}
+
+interface Props {
+  alertParams: Params;
+  setAlertParams: (key: string, value: any) => void;
+  setAlertProperty: (key: string, value: any) => void;
+}
+
+export function TransactionDurationAnomalyAlertTrigger(props: Props) {
+  const { setAlertParams, alertParams, setAlertProperty } = props;
+  const { urlParams } = useUrlParams();
+  const transactionTypes = useServiceTransactionTypes(urlParams);
+  const { serviceName, start, end } = urlParams;
+  const { environmentOptions } = useEnvironments({ serviceName, start, end });
+
+  if (!transactionTypes.length || !serviceName) {
+    return null;
+  }
+
+  const defaults: Params = {
+    windowSize: 15,
+    windowUnit: 'm',
+    transactionType: transactionTypes[0],
+    serviceName,
+    environment: urlParams.environment || ALL_OPTION.value,
+    anomalyScore: 75,
+  };
+
+  const params = {
+    ...defaults,
+    ...alertParams,
+  };
+
+  const fields = [
+    <EuiExpression
+      description={i18n.translate(
+        'xpack.apm.transactionDurationAnomalyAlertTrigger.service',
+        {
+          defaultMessage: 'Service',
+        }
+      )}
+      value={
+        <EuiText className="eui-displayInlineBlock">
+          <h5>{serviceName}</h5>
+        </EuiText>
+      }
+    />,
+    <PopoverExpression
+      value={
+        params.environment === ALL_OPTION.value
+          ? ALL_OPTION.text
+          : params.environment
+      }
```

Comment:
Can we use `[getEnvironmentLabel](https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/common/environment_filter_values.ts#L12)`?

```suggestion
      value={getEnvironmentLabel(params.environment)}
```

Reply:
Btw this is repeated in 

https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/public/components/shared/ErrorRateAlertTrigger/index.tsx#L54-L58

and 

https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/public/components/shared/TransactionDurationAlertTrigger/index.tsx#L67-L71

Might make sense to update there as well.

Reply:
While you are at it:
can you move [`ALL_OPTION`](https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/public/hooks/useEnvironments.tsx#L16-L21) and [`NOT_DEFINED_OPTION`](https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/public/components/shared/EnvironmentFilter/index.tsx#L35-L40) to 

https://github.com/elastic/kibana/blob/f61df057727c4adf0aa4bd1e72f108ec23945567/x-pack/plugins/apm/common/environment_filter_values.ts#L8-L14 ?